include/dlaf/util_matrix.h

```diff
@@ -98,11 +105,26 @@ bool multipliable_sizes(common::Size2D<IndexT, Tag> a, common::Size2D<IndexT, Ta
 }
 
 /// Returns true if matrices `a`, `b` and `c` have matrix multipliable sizes and block sizes.
-template <class T, Device D>
-bool multipliable(const Matrix<const T, D>& a, const Matrix<const T, D>& b, const Matrix<const T, D>& c,
-                  const blas::Op opA, const blas::Op opB) noexcept {
-  return multipliable_sizes(a.size(), b.size(), c.size(), opA, opB) &&
-         multipliable_sizes(a.blockSize(), b.blockSize(), c.blockSize(), opA, opB);
+template <template <class, Device> class MatrixLikeA, template <class, Device> class MatrixLikeB,
+          template <class, Device> class MatrixLikeC, class T, Device D>
+bool multipliable(const MatrixLikeA<const T, D>& a, const MatrixLikeB<const T, D>& b,
+                  const MatrixLikeC<const T, D>& c, const blas::Op opA, const blas::Op opB) noexcept {
+  const bool isSizeOk = multipliable_sizes(a.size(), b.size(), c.size(), opA, opB);
+
+  if (a.size().isEmpty() || c.size().isEmpty())
+    return isSizeOk;
+
+  const bool allSameGrid = same_process_grid(a, b) && same_process_grid(b, c);
+  const bool isTileSizeOk =
+      multipliable_sizes(a.baseTileSize(), b.baseTileSize(), c.baseTileSize(), opA, opB);
+  const bool isOffsetOk =
+      multipliable_sizes(a.tileSize({0, 0}), b.tileSize({0, 0}), c.tileSize({0, 0}), opA, opB);
+
+  if (local_matrix(c))
+    return allSameGrid && isSizeOk && isTileSizeOk && isOffsetOk;
+
+  // TODO distributed (fix offset)
```

Comment:
What exactly is this about?

Reply:
This is because in this PR we are contextually refactoring also `multipliable` assertions, and this does not fully cover the distributed case (specifically, what it is going to be implemented in #1022)